panel/param.py

```diff
@@ -802,10 +801,15 @@ def __init__(self, object=None, **params):
 
     @param.depends('object', watch=True)
     def _validate_object(self):
-        dependencies = getattr(self.object, '_dinfo', None)
+        if not callable(self.object):
+            return
+        dependencies = getattr(self.object, '_dinfo', {})
         if not dependencies or not dependencies.get('watch'):
             return
-        fn_type = 'method' if type(self) is ParamMethod else 'function'
+        if isinstance(self, ParamMethod):
+            fn_type = 'method'
```

Comment:
`fn_type` should probably be a class variable.

Reply:
Can `ParamRef` only accept a `function` or any `param.rx` object? If it is the latter, it could have a custom `fn_type`.

Reply:
It can accept any valid reference (and param allows you to register custom references), `fn_type` makes no sense to me in that context.

Reply:
I think your objection comes from the fact that I'm reference subclasses in this method and I agree that adding a fn_type would be a way out of that but I'm probably just going to move this validation logic instead.